### PR TITLE
Add pool video

### PR DESCRIPTION
**Pull request by darkterreur**
there are : 
- PoolVideo CRUD
- Video CRUD
- Link between PoolVideo and Video

### CODE REVIEW

File: src/AppBundle/Repository/PoolVideoRepository.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+use Doctrine\ORM\EntityRepository;
+
+
+
+class PoolVideoRepository extends EntityRepository
+{
+
+    public function findAllWithCountVideo()
+    {
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Repository/PoolVideoRepository.php]: To remove

File: src/UserBundle/Form/SubscribeType.php [hunk 2]
```diff
@@ -15,12 +16,19 @@
...
 {
-	public function buildform(FormBuilderInterface $builder, array $option)
-	{
-		$builder->add('username', TextType::class)
+    public function buildform(FormBuilderInterface $builder, array $option)
+    {
+        $builder->add('username', TextType::class)
                 ->add('password', PasswordType::class)
                 ->add('email',    EmailType::class)
                // ->add('save',     SubmitType::class)
```
Comment (geoffrey-chameroy) [inline src/UserBundle/Form/SubscribeType.php]: Remove useless comment

File: src/UserBundle/Resources/views/Security/login.html.twig [hunk 3]
```diff
@@ -9,18 +9,34 @@
         <div class="alert alert-danger">{{ error.message }}</div>
     {% endif %}
 
-    <div class="well">
+    <div class="">
         {# Le formulaire, avec URL de soumission vers la route « login_check » comme on l'a vu #}
```
Comment (geoffrey-chameroy) [inline src/UserBundle/Resources/views/Security/login.html.twig]: Remove useless comment

File: src/AppBundle/Entity/Video.php [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+     * Set poolVideo
+     *
+     * @param \stdClass $poolVideo
+     *
+     * @return Video
+     */
+    public function setPoolVideo($poolVideo)
+    {
+        
+        //$poolVideo->addVideo($this);
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Entity/Video.php]: Remove useless comment

File: src/AppBundle/Controller/BackendController.php [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+        dump($video);
+
+        $form = $this->createForm(ModifVideoType::class, $video);
+
+        if ($request->isMethod('POST')) {
+            $form->handleRequest($request);
+
+            if ($form->isValid()) {
+                //$em = $this->getDoctrine()->getManager();
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Controller/BackendController.php]: Remove useless comment

File: src/AppBundle/Controller/BackendController.php [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        if ($request->isMethod('POST')) {
+            $form->handleRequest($request);
+
+            if ($form->isValid()) {
+                $em = $this->getDoctrine()->getManager();
+
+                $em->persist($video);
+                $em->flush();
+
+               // die;
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Controller/BackendController.php]: Remove useless comment

File: src/AppBundle/Controller/BackendController.php [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+
+
+    public function manageVideoAction()
+    {
+        $listVideo = $this->getDoctrine()->getManager()->getRepository('AppBundle:Video')->findAll();
+
+        return $this->render('Backend/listeVideo.html.twig',array(
+                        'listVideo' => $listVideo,
+                    ));
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Controller/BackendController.php]: Bad indentation

File: src/AppBundle/Controller/BackendController.php [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+            throw new NotFoundHttpException("Error Pool Video with id ".$id." don't exist.");
+        }
+
+        $form = $this->createForm(PoolVideoType::class, $poolVideo);
+
+        if ($request->isMethod('POST')) {
+            $form->handleRequest($request);
+
+            if ($form->isValid()) {
+                //$em = $this->getDoctrine()->getManager();
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Controller/BackendController.php]: Remove useless comment

File: src/AppBundle/Controller/BackendController.php [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+
+class BackendController extends Controller
+{
+    public function managePoolVideoAction()
+    {
+        $listPoolVideo = $this->getDoctrine()->getManager()->getRepository('AppBundle:PoolVideo')->findAll();
+
+        return $this->render('Backend/listePoolVideo.html.twig',array(
+                        'listPoolVideo' => $listPoolVideo,
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Controller/BackendController.php]: Bad indentation

File: composer.json [hunk 10]
```diff
@@ -39,7 +39,8 @@
     "require-dev": {
         "sensio/generator-bundle": "^3.0",
         "symfony/phpunit-bridge": "^3.0",
-        "doctrine/doctrine-fixtures-bundle": "^2.3"
+        "doctrine/doctrine-fixtures-bundle": "^2.3",
+        "phpunit/phpunit": "3.7.*"
```
Comment (geoffrey-chameroy) [inline composer.json]: Why add it, if you don't make any tests ?

Comment (darkterreur) [inline composer.json]: exactly but i would like make test in this project. it's first step

File: src/AppBundle/Repository/PoolVideoRepository.php (hunk 1 above)
Comment (darkterreur) [inline src/AppBundle/Repository/PoolVideoRepository.php]: Ho no, can i keep this for example for me. I can consulte with git project. Please.

Comment (geoffrey-chameroy) [inline src/AppBundle/Repository/PoolVideoRepository.php]: No, keep it in your documents, but not in the repository.

File: composer.json (hunk 10 above)
Comment (geoffrey-chameroy) [inline composer.json]: ok ;)

File: src/AppBundle/Entity/Video.php [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+     * @ORM\PrePersist()
+     * @ORM\PreUpdate()
+     */
+    public function preUpload()
+    {
+        if (null === $this->file) {
+            return;
+        }
+
+        $this->url = $this->file->guessExtension();
```
Comment (alcalyn) [inline src/AppBundle/Entity/Video.php]: This `url` field is receiving the file extension, and not the url to the uploaded file as I expect.

I tested it, and I got the video extension in the url field in database.

File: src/AppBundle/Controller/BackendController.php [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+    }
+
+
+    public function modifVideoAction(Request $request, $id)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $video = $em->getRepository('AppBundle:Video')
+            ->find($id);
+
+        dump($video);
```
Comment (geoffrey-chameroy) [inline src/AppBundle/Controller/BackendController.php]: to remove

File: .gitignore [hunk 13]
```diff
@@ -44,5 +44,9 @@
 # Composer PHAR
 /composer.phar
 
+# php-cs-fixer PHAR
+/php-cs-fixer.phar
+.php_cs.cache
+
```
Comment (geoffrey-chameroy) [inline .gitignore]: To remove

**Pull request merged**